klock/src/commonMain/kotlin/com/soywiz/klock/KlockLocalContext.kt

```diff
@@ -0,0 +1,16 @@
+package com.soywiz.klock
+
+data class KlockLocalContext(val gender: KlockLocaleGender) {
```

Comment:
Maybe this could be somehow private or internal, so people can construct this directly? Since I guess adding new parameters to a constructor is not ABI-compatible, but the withGender should keep it ABI-compatible if at some point other language requires other extra context.

We could also asume no ABI compatibility here and only source-code compatibility, and either force named arguments, or just add default values, and add new parameters always at the end.
In that case, maybe we don't need the .withGender, and we can just use the `data class` `KlockLocaleContext(...).copy(gender = ...)`. What do you think?

Reply:
Good point.

I think if we want to deal with ABI-compatibility, `KlockLocaleContext(...).copy(gender = ...)` can become incompatible too... Do you think it's true? I've just googled briefly and haven't found info about compatibility of data classes. If yes, then I guess we should restrict `copy` from external usage too, for example, by making this class not-data.

If you mean only source-compatibility, then copy shouldn't be the problem.

Anyways, I agree that we should restrict visibility of the constructor. I'll do it.

Reply:
@soywiz, so which variant is better?

1.

```kt
data class KlockLocalContext internal constructor(val gender: KlockLocaleGender) {

    companion object {

        val empty = KlockLocalContext(KlockLocaleGender.Neuter)
    }
}
```

2.

```
data class KlockLocalContext(val gender: KlockLocaleGender = KlockLocaleGender.Neuter)
```

The last one, right?

Reply:
I would go with 2 because of simplicity. And it can be source-level compatible, so 👍